### ISSUE TITLE
400 (Bad request) on accepting or rejecting a service provider to a game

### ISSUE THREAD

**Issue by tarekmorsi** [SEGUC17/codeologists#116]
1- Severity --> Medium
2- Reported by --> Tarek Morsi

3- Upon accepting a service provider to a game, an alert is displayed with an unexplained message and the application crashes (any type of action afterwards, like routing or clicking any button displays a colossal amount of errors in the console).

![screenshot 79](https://cloud.githubusercontent.com/assets/25376323/25591002/a3750cf2-2e67-11e7-8610-f16201b10028.png)

![screenshot 80](https://cloud.githubusercontent.com/assets/25376323/25591007/aab5f378-2e67-11e7-9413-2bec0e0a603c.png)

4- Steps to Produce The Issue:
a. open the website
b. sign in as a service provider
c. go to /games and request to join a game.
d. sign out and sign back in as a player
e. accept the service provider to the game.

5- Expected to be accepted to the game and be shown a message that the request was accepted or rejected (in case of an error, an error should be displayed.)

**Comment by mahmoud-adel-mahmoud** [comment]
It seems to be that the server isn't crashed

**Comment by minaamir26** [comment]
The title should be descriptive, the app does not crash it is working normally
The error gets back from the backend, you can see it in the response in the Networks tab (the error code is 400) but its message is not displayed, that's it all
The severity is not high as written, you can see the doctor's documentation on slack channel for the severity

**Comment by tarekmorsi** [comment]
My mistake, let me rephrase what I was saying and back my words up with a few more screenshots.

**Comment by minaamir26** [comment]
It is not an error by the way, we are who sending this message with this code in the backend, our only mistake was just we had not displayed the message to the user

**Comment by mahmoud-adel-mahmoud** [comment]
Excuse me, but the severity is now medium not high as mentioned

**Comment by El-Etreby** [comment]
<img width="1280" alt="screen shot 2017-05-01 at 9 38 07 pm" src="https://cloud.githubusercontent.com/assets/25392406/25591910/c892fcf2-2eb6-11e7-8329-ebf9f43663d4.png">
<img width="1280" alt="screen shot 2017-05-01 at 9 38 13 pm" src="https://cloud.githubusercontent.com/assets/25392406/25591908/c86919b4-2eb6-11e7-945c-0e1277f0858c.png">
<img width="1280" alt="screen shot 2017-05-01 at 9 38 35 pm" src="https://cloud.githubusercontent.com/assets/25392406/25591909/c86b0440-2eb6-11e7-9310-c6c352ffa194.png">

I do think that severity should be high since the application did crash when i tried the same steps, most probably that issue occurs whenever the user accepts the game requested by a provider.

**Comment by minaamir26** [comment]
**Application crashes** means that : you will not be able to open the website at all after the bug is detected :D but if you go to the website again and type the url, you will be able to continue

**Comment by El-Etreby** [comment]
According to the announcement on slack "severe: causes the application crashes- application generates an incorrect response
medium: the user is left with no feedback after successful completion of an operation. A transaction is completed yet incurs considerable delay." I am pretty sure that this issue isnt medium, in fact it is severe since the application generates an incorrect response whenever the user accepts the game requested by the provider.

**Comment by mahmoud-adel-mahmoud** [comment]
Generating wrong response means redirecting you to a wrong page.
But our case is an error confusion.
